lib/active_fedora/file.rb

```diff
@@ -63,7 +63,15 @@ def ldp_connection
     # If this file has a parent with ldp#contains, we know it is not new.
     # By tracking exists we prevent an unnecessary HEAD request.
     def new_record?
-      !@exists && ldp_source.new?
+      !@exists && ldp_new_record?
```

Comment:
@hectorcorrea try rebasing against https://github.com/projecthydra/active_fedora/pull/840. Then would you be able to revert your change to new_record?